fathom/src/surface/elaboration.rs

```diff
@@ -1886,10 +1886,113 @@ impl<'interner, 'arena, 'error> Context<'interner, 'arena, 'error> {
 
                 (core::Term::FormatOverlap(labels, formats), format_type)
             }
+            Term::BinOp(range, lhs, op, rhs) => {
+                // de-sugar into function application
+                let (lhs_expr, lhs_type) = self.synth(lhs);
+                let (rhs_expr, rhs_type) = self.synth(rhs);
+                let lhs_type = self.elim_context().force(&lhs_type);
+                let rhs_type = self.elim_context().force(&rhs_type);
+                let operand_types =
+                    Option::zip(lhs_type.match_prim_spine(), rhs_type.match_prim_spine());
+                let (fun, output_type) = self.match_bin_op(
+                    *range,
+                    lhs.range(),
+                    rhs.range(),
+                    &lhs_type,
+                    &rhs_type,
+                    *op,
+                    operand_types,
+                );
+                let fun_app = core::Term::FunApp(
+                    self.scope.to_scope(core::Term::FunApp(
+                        self.scope.to_scope(fun),
+                        self.scope.to_scope(lhs_expr),
+                    )),
+                    self.scope.to_scope(rhs_expr),
+                );
+
+                (fun_app, output_type)
```

Comment:
Might be cleaner to move this all into some `match_bin_op`, perhaps renaming to `synth_bin_op`